azurerm/internal/services/search/search_service_data_source_test.go

```diff
@@ -0,0 +1,46 @@
+package search_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/acceptance/check"
+)
+
+type SearchServiceDataSource struct {
+}
+
+func TestAccDataSourceSearchService_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "data.azurerm_search_service", "test")
+	r := SearchServiceDataSource{}
+
+	data.DataSourceTest(t, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).Key("replica_count").Exists(),
+				check.That(data.ResourceName).Key("partition_count").Exists(),
+				check.That(data.ResourceName).Key("primary_key").Exists(),
+				check.That(data.ResourceName).Key("secondary_key").Exists(),
+				check.That(data.ResourceName).Key("public_network_access_enabled").Exists(),
+				check.That(data.ResourceName).Key("identity.type").Exists(),
```

Comment:
this is a list so this needs to be:

```suggestion
				check.That(data.ResourceName).Key("identity.0.type").Exists(),
```

to access the element - else the test fails:

```
------- Stdout: -------
=== RUN   TestAccDataSourceSearchService_basic
=== PAUSE TestAccDataSourceSearchService_basic
=== CONT  TestAccDataSourceSearchService_basic
testing.go:684: Step 0 error: Check failed: Check 6/8 error: data.azurerm_search_service.test: Attribute 'identity.type' expected to be set
--- FAIL: TestAccDataSourceSearchService_basic (108.32s)
FAIL
```

but if we can fix that up then this should otherwise be good to go 👍